rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -606,8 +621,15 @@ void SqliteStorage::fill_topics_and_types()
         std::string, std::string, std::string, std::string>();
 
       for (auto result : query_results) {
+        auto yaml_node = YAML::Load(std::get<3>(result));
```

Comment:
Here, a conversion from serialized YAML (`std::string`) & `std::vector<rclcpp::QoS>` is needed.